bull-transformer-generator/transformer-bytecode-adapter/src/main/java/com/hotels/beans/generator/bytecode/TransformerBytecodeAdapter.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (C) 2019-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hotels.beans.generator.bytecode;
+
+import com.hotels.beans.generator.core.Transformer;
+import com.hotels.beans.generator.core.TransformerSpec;
+import com.squareup.javapoet.JavaFile;
+
+import lombok.Builder;
+import lombok.extern.slf4j.Slf4j;
+
+import net.openhft.compiler.CachedCompiler;
+import net.openhft.compiler.CompilerUtils;
+
+/**
+ * A bytecode adapter for {@link TransformerSpec} that compiles at runtime the source code
+ * represented by the model, and creates a new instance of {@link Transformer}.
+ * <p>
+ * This component is intended for internal usage in Bull as it's a low-level wrapper for
+ * runtime compilation of models. Clients should prefer higher level interfaces
+ * such as {@code TransformerRegistry}.
+ */
+@Builder
+@Slf4j
+public final class TransformerBytecodeAdapter {
+
+    /**
+     * The default Transformer package if none is specified.
+     */
+    static final String DEFAULT_PACKAGE = "com.hotels.beans.transformer";
+
+    /**
+     * The Transformer model for generating source code.
+     */
+    private final TransformerSpec spec;
+
+    /**
+     * The Java package to use for every generated Transformer source code.
+     */
+    @Builder.Default
+    private final String packageName = DEFAULT_PACKAGE;
+
+    /**
+     * The compiler instance to use at runtime.
+     */
+    @Builder.Default
+    private final CachedCompiler compiler = CompilerUtils.CACHED_COMPILER;
+
+    /**
+     * Dynamically create a new {@link Transformer} instance.
+     * @param <A> the type parameter
+     * @param <B> the type parameter
+     * @param source the source
+     * @param destination the destination
+     * @return the transformer
+     */
+    @SuppressWarnings("Unchecked Assignment")
```

Comment:
this suppression does not work, please replace it with:
```suggestion
    @SuppressWarnings("unchecked")
```

Reply:
done

Reply:
nice catch